juju/juju
# [JUJU-3394] Record the jwt auth token along side te macaroon on the cmr app entity

As a first step in allowing cross model relations to work with JAAS auth tokens, this PR records the auth token on the remote application. The macaroon is stored there - we need to now add the auth token as well. Only one of these will be set.

The macaroon or auth token is used when making cross controller cmr calls. There are several of these api calls. This PR prototypes the register remote relation call. The token is checked but there's no refresh yet if it is invalid.

The cross model auth is reworked to handle both macaroon and token checks.

As a drive by, remove the user in db check for model owner. This is checked in the facade anyway and we are trying to move away from a model owner attribute, and it also breaks jaas rebac.

## QA steps

Check that cross model relations still works for the juju cli. JAAS will validate the auth token stuff.
add a model and deploy juju-qa-dummy-source and offer
add a model and deploy juju-qa-dummy-sink and consume the offer and relate
add a user and grant consume on the offer and add-model on the controller cloud
login as that user and add a model and consume and relate the offer
